packages/upload-client/test/blob.test.js

```diff
@@ -0,0 +1,702 @@
+import assert from 'assert'
+import * as Client from '@ucanto/client'
+import * as Server from '@ucanto/server'
+import { provide } from '@ucanto/server'
+import * as CAR from '@ucanto/transport/car'
+import * as Signer from '@ucanto/principal/ed25519'
+import * as UCAN from '@web3-storage/capabilities/ucan'
+import * as BlobCapabilities from '@web3-storage/capabilities/blob'
+import * as Blob from '../src/blob.js'
+import { serviceSigner } from './fixtures.js'
+import { randomCAR } from './helpers/random.js'
+import { mockService } from './helpers/mocks.js'
+import {
+  validateAuthorization,
+  setupBlobAddSuccessResponse,
+  setupBlobAdd4xxResponse,
+  setupBlobAdd5xxResponse,
+} from './helpers/utils.js'
+import { fetchWithUploadProgress } from '../src/fetch-with-upload-progress.js'
+
+describe('Blob.add', () => {
+  it('stores a DAG with the service', async () => {
+    const space = await Signer.generate()
+    const agent = await Signer.generate()
+    const car = await randomCAR(128)
```

Comment:
I'd use just random bytes here - there's no need to associate this with CARs. Same for all of these tests.

Reply:
addressed